Link/Scaffolding/ScaffoldingFactory.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using PCL.Core.Link.Scaffolding.EasyTier;
+using PCL.Core.Link.Scaffolding.Models;
+using PCL.Core.Link.Scaffolding.Server;
+using PCL.Core.Net;
+
+namespace PCL.Core.Link.Scaffolding;
+
+public static class ScaffoldingFactory
+{
+    public static async Task<ScaffoldingClient> CreateClientAsync(string playerName, string roomCode, RoomType from)
+    {
+        var machineId = Utils.Secret.Identify.LaunchId;
+
+        if (!RoomCodeGenerator.TryParse(roomCode, out var info))
+        {
+            throw new ArgumentException("Invalid room share code.", nameof(roomCode));
+        }
+
+        var entity = _CreateEasyTierEntity(info, 0, 0, false);
+        entity.Launch();
+        var players = await entity.GetPlayersAsync().ConfigureAwait(false);
+        var hostInfo = players.Players?.FirstOrDefault();
+
+        if (hostInfo is null)
+        {
+            throw new ArgumentNullException(nameof(hostInfo), "Can not get the host information.");
+        }
+
+        var host = from switch
+        {
+            RoomType.PCLCE => "10.114.51.41",
+            RoomType.Terracotta => "10.144.144.1",
+            _ => throw new ArgumentOutOfRangeException(nameof(from), from, "Unsupported room type.")
+        };
+
+        if (!int.TryParse(hostInfo.HostName[22..], out var scfPort))
+        {
+            throw new ArgumentException("Invalid hostname.", nameof(hostInfo));
+        }
+
+        return new ScaffoldingClient(host, scfPort, playerName, machineId, "pcl2-ce");
```

Comment:
```suggestion
        return new ScaffoldingClient(host, scfPort, playerName, machineId, $"PCL CE {启动器版本}, EasyTier {ET 版本}");
```

应尽可能使用 `PCL CE` 表述

目前似乎还没法从 Core 拿启动器版本，可能需要加个常量？